Makefile.include

```diff
@@ -454,12 +454,20 @@ else
 ifeq (,$(RIOTNOLINK))
 link: ..compiler-check ..build-message $(ELFFILE) $(FLASHFILE) $(HEXFILE) print-size
 else
-link: ..compiler-check ..build-message $(BASELIBS)
+link: ..compiler-check ..build-message $(ELFFILE).nolink
 endif # RIOTNOLINK
 
 $(ELFFILE): $(BASELIBS)
 	$(Q)$(_LINK) -o $@
 
+# Try to link and ensure it fails
+$(ELFFILE).nolink: FORCE
+$(ELFFILE).nolink: $(BASELIBS)
+	$(Q)\
+	  $(_LINK) -o $(ELFFILE) 2>&1 > $@; \
```

Comment:
I left `ELFFILE` on purpose, because you could check the elffile if it succeeds and you would like to analyze it. All the files from `BASELIBS` are also still there.

But I can change it if wanted.